src/librustc_errors/emitter.rs

```diff
@@ -908,7 +912,8 @@ impl EmitterWriter {
         // Print out the annotate source lines that correspond with the error
         for annotated_file in annotated_files {
             // we can't annotate anything if the source is unavailable.
-            if annotated_file.file.src.is_none() {
+            if annotated_file.file.src.is_none()
+                    && annotated_file.file.external_src.borrow().is_absent() {
```

Comment:
We can't do that, since said method would have to be defined in a place where the `FileLoader` doesn't exist.